app/root.tsx

```diff
@@ -23,22 +24,26 @@ import {
 import Footer from './components/footer';
 import SignInButton from './components/sign-in-button';
 
-import { withAuth, getSignInUrl, signOut } from '@workos-inc/authkit-remix';
+import {
+  getSignInUrl,
+  signOut,
+  authkitLoader,
+} from '@workos-inc/authkit-remix';
 
 export const links: LinksFunction = () => [
   ...(cssBundleHref ? [{ rel: 'stylesheet', href: cssBundleHref }] : []),
 ];
 
-export async function loader({ request }: LoaderFunctionArgs) {
-  const data = await withAuth(request, {
-    debug: true,
-  });
-
-  return json({
-    authorizationUrl: await getSignInUrl(),
-    ...data,
-  });
-}
+export const loader: LoaderFunction = (args: LoaderFunctionArgs) =>
```

Comment:
```suggestion
export const loader = (args: LoaderFunctionArgs) =>
```
Pretty sure that typing this as `LoaderFunction` will prevent type inference from working with `useLoaderData`. Typing the function parameter should be sufficient.